shelley-ma/impl/src/Cardano/Ledger/Allegra/Translation.hs

```diff
@@ -0,0 +1,110 @@
+
+instance Crypto c => TranslateEra (AllegraEra c) ShelleyState where
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) Tx where
+  type TranslationError (AllegraEra c) Tx = ()
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) ShelleyGenesis where
+  translateEra ctxt genesis =
+    return
+      ShelleyGenesis
+        { sgSystemStart = sgSystemStart genesis,
+          sgNetworkMagic = sgNetworkMagic genesis,
+          sgNetworkId = sgNetworkId genesis,
+          sgActiveSlotsCoeff = sgActiveSlotsCoeff genesis,
+          sgSecurityParam = sgSecurityParam genesis,
+          sgEpochLength = sgEpochLength genesis,
+          sgSlotsPerKESPeriod = sgSlotsPerKESPeriod genesis,
+          sgMaxKESEvolutions = sgMaxKESEvolutions genesis,
+          sgSlotLength = sgSlotLength genesis,
+          sgUpdateQuorum = sgUpdateQuorum genesis,
+          sgMaxLovelaceSupply = sgMaxLovelaceSupply genesis,
+          sgProtocolParams = translateEra' ctxt (sgProtocolParams genesis),
+          sgGenDelegs =
+            Map.fromList
+              . map (bimap (translateEra' ctxt) (translateEra' ctxt))
+              . Map.toList
+              . sgGenDelegs
+              $ genesis,
+          sgInitialFunds = Map.mapKeys (translateEra' ctxt) $ sgInitialFunds genesis,
+          sgStaking = translateEra' ctxt $ sgStaking genesis
+        }
+
+--------------------------------------------------------------------------------
+-- The following instances are also needed by consensus, but only temporarily.
+-- These types should not be parameterised by the era but by the crypto. When
+-- this has changed (CAD-2000), the instances below can be removed.
+--------------------------------------------------------------------------------
+
+instance Crypto c => TranslateEra (AllegraEra c) HashHeader where
+  translateEra _ (HashHeader hash) = return $ HashHeader $ Hash.castHash hash
+
+instance Crypto c => TranslateEra (AllegraEra c) ChainDepState where
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) LedgerView where
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) (VKey 'BlockIssuer) where
+  translateEra _ = undefined
+
+--------------------------------------------------------------------------------
+-- Auxiliary instances and functions
+--------------------------------------------------------------------------------
+
+instance Crypto c => TranslateEra (AllegraEra c) (PParams' Identity) where
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) (KeyHash 'Genesis) where
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) GenDelegPair where
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) ShelleyGenesisStaking where
+  translateEra _ = undefined
+
+instance Crypto c => TranslateEra (AllegraEra c) Addr where
+  translateEra _ = undefined
```

Comment:
The instance for `ShelleyGenesis` needed these auxiliary instances. When CAD-2000 is implemented, some of them can go (`KeyHash 'Genesis`, `GenDelegPair`).